src/app/views/query-response/QueryResponse.tsx

```diff
@@ -17,6 +19,13 @@ class QueryResponse extends Component<IQueryResponseProps, {}> {
     super(props);
   }
 
+  public componentDidUpdate(nextProps: any) {
+    const { scopes } = this.props;
+    if (nextProps.scopes !== scopes) {
+      this.props.actions!.getConsent();
```

Comment:
Could you add a check if the new scopes are empty? There would be no need to get consent on an empty set of scopes